### PR TITLE
Cosmetic autoconsent rules

### PR DESCRIPTION
**Pull request by muodov**
Task/Issue URL: https://app.asana.com/0/72649045549333/1203672644429320/f

**we should first release [Privacy Dashboard](https://github.com/duckduckgo/privacy-dashboard/pull/21) before merging this PR**

### Description

This adds support for the latest autoconsent version with cosmetic rules.

Related Privacy Dashboard PR: https://github.com/duckduckgo/privacy-dashboard/pull/21

### Steps to test this PR

1. enable the CPM feature
2. use VPN if you're not in Europe (cookie popups are rare in the US)
1. go to https://w3resource.com or https://bitcoin.org -> the animation should say "pop-up hidden" (cosmetic rule)
1. go to https://etsy.com or https://doodle.com/ -> the animation should say "pop-up managed"
2. Open Privacy Dashboard and click on the autoconsent-related row
3. on the next screen, click "Change in Settings" -> this should open the CPM settings screen
1. observe the changes in CPM feature description

### Animation
| Before  | After (cosmetic rule) | After (normal rule) |
| ------ | ----- | ---- |
| animation-managed | animation-hidden | animation-managed |

### CPM settings
| Before  | After |
| ------ | ----- |
|settings-before|settings-new|

### Dashboard
| Before  | After (non-cosmetic, 1) | After (non-cosmetic, 2) |
| ------ | ----- | ----- |
| dashboard-before | dashboard-after-1 | dashboard-after-2 |

| Before  | After (cosmetic, 1) | After (cosmetic, 2) |
| ------ | ----- | ----- |
| dashboard-before | dashboard-after-3 | dashboard-after-4|

### CODE REVIEW

File: privacy-dashboard/privacy-dashboard-impl/src/main/java/com/duckduckgo/privacy/dashboard/impl/ui/PrivacyDashboardHybridViewModel.kt
Comment (cmonfortep) [inline privacy-dashboard/privacy-dashboard-impl/src/main/java/com/duckduckgo/privacy/dashboard/impl/ui/PrivacyDashboardHybridViewModel.kt:64]: I would renamed this command to `OpenAutoconsentSettings` to avoid confusion with Settings root.

File: privacy-dashboard/privacy-dashboard-impl/src/main/java/com/duckduckgo/privacy/dashboard/impl/ui/PrivacyDashboardPayloadAdapter.kt
Comment (cmonfortep) [inline privacy-dashboard/privacy-dashboard-impl/src/main/java/com/duckduckgo/privacy/dashboard/impl/ui/PrivacyDashboardPayloadAdapter.kt:38]: what's inside `payload`? it seems nothing we use. I see we are mapping the payload into `SettingsPayload`, but when opening the screen, no value is required. Maybe we don't need any logic here?

File: privacy-dashboard/privacy-dashboard-impl/src/test/java/com/duckduckgo/privacy/dashboard/impl/ui/PrivacyDashboardRendererTest.kt
Comment (cmonfortep) [inline privacy-dashboard/privacy-dashboard-impl/src/test/java/com/duckduckgo/privacy/dashboard/impl/ui/PrivacyDashboardRendererTest.kt:87]: can you also add the call that is triggered?

I assume:

`assertNotNull(captor.allValues.find { it.startsWith("javascript:onChangeConsentManaged") })`

File: privacy-dashboard/privacy-dashboard-impl/src/main/java/com/duckduckgo/privacy/dashboard/impl/ui/PrivacyDashboardRendererFactory.kt
Comment (cmonfortep) [inline privacy-dashboard/privacy-dashboard-impl/src/main/java/com/duckduckgo/privacy/dashboard/impl/ui/PrivacyDashboardRendererFactory.kt:57]: NIT: Same as with the viewmodel `Command`, could we rename to `onOpenAutoconsentSettings`?

File: privacy-dashboard/privacy-dashboard-impl/src/main/java/com/duckduckgo/privacy/dashboard/impl/ui/PrivacyDashboardRenderer.kt
Comment (cmonfortep) [inline privacy-dashboard/privacy-dashboard-impl/src/main/java/com/duckduckgo/privacy/dashboard/impl/ui/PrivacyDashboardRenderer.kt:72]: I think we can remove the comment here.

File: package.json
Comment (cmonfortep) [inline package.json:20]: what's the plan here? I assume we want to use a release version before merging?

**Review by cmonfortep — COMMENTED** [review]
Reviewed the code inside privacy dashboard submodules. Only minor comments, looks good to me. Great job @muodov

File: package.json
Comment (aitorvs) [reply package.json:20]: +1, we can't merge this PR pointing to anything that is not `duckduckgo` repo

Comment (muodov) [reply package.json:20]: absolutely, I am going to replace https://github.com/duckduckgo/Android/pull/2738/commits/e884025017e089fbf1d2029866093e1acb9d36eb once the corresponding [dashboard PR](https://github.com/duckduckgo/privacy-dashboard/pull/21) is merged

File: privacy-dashboard/privacy-dashboard-impl/src/main/java/com/duckduckgo/privacy/dashboard/impl/ui/PrivacyDashboardPayloadAdapter.kt
Comment (muodov) [reply privacy-dashboard/privacy-dashboard-impl/src/main/java/com/duckduckgo/privacy/dashboard/impl/ui/PrivacyDashboardPayloadAdapter.kt:38]: Looking closely, I think @shakyShane's original intention was to make this method generic, and the `payload` would clarify which particular setting we'd like to open. So I [changed the logic](https://github.com/duckduckgo/Android/pull/2738/commits/7203f8b8fca575228c4bc5323f99a50a97c0fe83) a bit. This also addresses the your remarks about naming

File: privacy-dashboard/privacy-dashboard-impl/src/test/java/com/duckduckgo/privacy/dashboard/impl/ui/PrivacyDashboardRendererTest.kt
Comment (muodov) [reply privacy-dashboard/privacy-dashboard-impl/src/test/java/com/duckduckgo/privacy/dashboard/impl/ui/PrivacyDashboardRendererTest.kt:87]: Yup, added it in the last commit

File: autoconsent/autoconsent-impl/src/main/java/com/duckduckgo/autoconsent/impl/handlers/InitMessageHandlerPlugin.kt
Comment (marcosholgado) [inline autoconsent/autoconsent-impl/src/main/java/com/duckduckgo/autoconsent/impl/handlers/InitMessageHandlerPlugin.kt:79]: Is this (enableCosmeticRules) something we plan to add to the remote config at some point?

File: package.json
Comment (marcosholgado) [inline package.json:20]: Should be pointing to the ddg repo

**Review by marcosholgado — CHANGES_REQUESTED** [review]
@muodov this is pretty good. I left a few comments but nothing big. Just blocking the merge until the Ship Review is done and the packages.json points to the correct repos once the other pieces are ready.

File: autoconsent/autoconsent-impl/src/main/java/com/duckduckgo/autoconsent/impl/handlers/InitMessageHandlerPlugin.kt
Comment (muodov) [reply autoconsent/autoconsent-impl/src/main/java/com/duckduckgo/autoconsent/impl/handlers/InitMessageHandlerPlugin.kt:79]: There's no such plans at the moment. We can still disable single rules by name, that should be enough.

File: autoconsent/autoconsent-api/src/main/java/com/duckduckgo/autoconsent/api/AutoconsentNav.kt
Comment (muodov) [reply autoconsent/autoconsent-api/src/main/java/com/duckduckgo/autoconsent/api/AutoconsentNav.kt:25]: I've refactored it a bit and added a comment

File: autoconsent/autoconsent-impl/src/test/java/com/duckduckgo/autoconsent/impl/handlers/OptOutAndAutoconsentDoneMessageHandlerPluginTest.kt
Comment (muodov) [reply autoconsent/autoconsent-impl/src/test/java/com/duckduckgo/autoconsent/impl/handlers/OptOutAndAutoconsentDoneMessageHandlerPluginTest.kt]: Good point, I've changed it

File: package.json
Comment (muodov) [reply package.json]: PD is released, I've pinned it to the production version

File: package.json
Comment (muodov) [reply package.json]: PD is released, I've pinned it to the production version

**Comment by marcosholgado** [comment]
@cmonfortep happy to merge this?

**Comment by cmonfortep** [comment]
I've just tested it a bit and looks good to me 👍 

There's one minor suggestion about renaming `onOpenSettings` and similar `Command` to `AutoconsentSettings`for clarity. But a part from that, we can merge.

**Review by marcosholgado — APPROVED** [review]
LGTM!

**Pull request merged**
